src/hooks/api/use-cards-client.ts

```diff
@@ -74,28 +74,26 @@ export function useCardsClient() {
   //////////////////////
 
   // POST: /Cards
-  async function addCard(card: Card, deckId: number): Promise<NewCardsResponse> {
+  async function addCard(card: DraftCard | Card, deckId: number): Promise<NewCardsResponse> {
```

Comment:
We are free to cast between `Card` and `DraftCard` because the only added fields are optional. 

I guess `DraftCard` is not the right word. This type applies to cards that are currently being edited because I am attaching the custom audio blobs directly to the cards. These blobs will only exist in the edit deck view. 

`updateCard` and `addCard` are both `DraftCards` because either could have the blobs attached to them. 

I was at first maintaining a separate structure to keep track of the custom audio blobs. But since they have to be combined to make the request, it made it a lot simpler to attach the blobs directly to the card. 

Would `UnsavedCard` be better?

Reply:
either name is fair to me with that in mind

that said, `addCards` should be `DraftCard[] | Card[]` then?

Reply:
ope, yeah you right. It should be fixed now